eos_data_distribution/ndn/base.py

```diff
@@ -115,6 +115,11 @@ def __init__(self, name, face=None):
         self._callbackCount = 0
         self._responseCount = 0
 
+        if auto: self.start()
```

Comment:
(GitHub didn't get my comment through? Leaving it again...)

This is broken since you chain up before you construct the rest of the object. To give an example, `Consumer` chains up first, then sets `self.pit`. But since `start` here calls `expressInterest` before `self.pit` is initialized, we'll fail when we try to run `self.pit[filename] = ...`

This is why I don't like `auto=True` as a paradigm -- constructor ordering is very, very tricky to get right and I'd prefer not to deal with it at all.